File: frontend/src/services/api/types/agents.ts
/**
 * Agent, AgentConversation, AgentDocument, and related conversation types.
 */

import type { ConversationStatus, ConversationMessage } from './common';

export type AgentCreationStatus = 'pending' | 'in_progress' | 'completed';
export type EffortLevel = 'low' | 'medium' | 'high' | 'max';
export type AgentLayer = 'backend' | 'frontend' | 'design' | 'analysis' | 'test' | 'management' | 'maintenance' | 'data' | 'mobile';
export type TeamMemberRole = 'leader' | 'member';

export interface AgentDocument {
  name: string;
  path?: string;
  type: 'file' | 'url' | 'inline';
  content?: string;
}

export interface Agent {
  id: string;
  name: string;
  description?: string;
  role?: string;
  goals?: string[];
  context?: string;
  backend_type: 'claude' | 'opencode' | 'gemini' | 'codex';
  enabled: number;
  skills?: string[];
  documents?: AgentDocument[];
  system_prompt?: string;
  creation_conversation_id?: string;
  creation_status: AgentCreationStatus;
  triggers?: string[];
  color?: string;
  icon?: string;
  model?: string;
  temperature?: number;
  tools?: string[];
  autonomous?: number;
  allowed_tools?: string[];
  preferred_model?: string;
  effort_level?: EffortLevel;
  layer?: AgentLayer;
  detected_role?: string;
  matched_skills?: string[];
  created_at?: string;
  updated_at?: string;
}

export interface AgentConversation {
  id: string;
  agent_id?: string;
  status: ConversationStatus;
  messages?: ConversationMessage[];
  messages_parsed?: ConversationMessage[];
  created_at?: string;
  updated_at?: string;
}

export interface SkillInfo {
  name: string;
  description: string;
  source_path?: string;
}

// Skill Conversation API
export interface SkillConversation {
  id: string;
  status: ConversationStatus;
  messages?: ConversationMessage[];
  messages_parsed?: ConversationMessage[];
  created_at?: string;
  updated_at?: string;
}

// Plugin Conversation API
export interface PluginConversation {
  id: string;
  status: ConversationStatus;
  messages?: ConversationMessage[];
  messages_parsed?: ConversationMessage[];
  created_at?: string;
  updated_at?: string;
}

// Hook Conversation API
export interface HookConversation {
  id: string;
  status: ConversationStatus;
  messages_parsed?: ConversationMessage[];
}

// Command Conversation API
export interface CommandConversation {
  id: string;
  status: ConversationStatus;
  messages_parsed?: ConversationMessage[];
}

// Rule Conversation API
export interface RuleConversation {
  id: string;
  status: ConversationStatus;
  messages_parsed?: ConversationMessage[];
}

export interface DesignConversationSummary {
  id: string;
  entity_type: string;
  entity_id: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/db/schema/_agents.py
"""Agents and agent_conversations DDL."""


def create_agent_tables(conn):
    # Agents table - AI agents with richer context than triggers
    conn.execute("""
        CREATE TABLE IF NOT EXISTS agents (
            id TEXT PRIMARY KEY,
            name TEXT NOT NULL,
            description TEXT,
            role TEXT,
            goals TEXT,
            context TEXT,
            backend_type TEXT NOT NULL DEFAULT 'claude',
            enabled INTEGER DEFAULT 1,
            skills TEXT,
            documents TEXT,
            system_prompt TEXT,
            creation_conversation_id TEXT,
            creation_status TEXT DEFAULT 'completed',
            triggers TEXT,
            color TEXT,
            icon TEXT,
            model TEXT,
            temperature REAL,
            tools TEXT,
            autonomous INTEGER DEFAULT 0,
            allowed_tools TEXT,
            layer TEXT,
            detected_role TEXT,
            matched_skills TEXT,
            preferred_model TEXT,
            effort_level TEXT DEFAULT 'medium',
            memory_config TEXT,
            created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
            updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
        )
    """)
    conn.execute("CREATE INDEX IF NOT EXISTS idx_agents_enabled ON agents(enabled)")
    conn.execute("CREATE INDEX IF NOT EXISTS idx_agents_created_at ON agents(created_at DESC)")

    # Agent conversations table - for agent creation flow
    conn.execute("""
        CREATE TABLE IF NOT EXISTS agent_conversations (
            id TEXT PRIMARY KEY,
            agent_id TEXT,
            status TEXT DEFAULT 'active',
            messages TEXT,
            created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
            updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
            FOREIGN KEY (agent_id) REFERENCES agents(id) ON DELETE SET NULL
        )
    """)
    conn.execute(
        "CREATE INDEX IF NOT EXISTS idx_agent_conversations_status ON agent_conversations(status)"
    )
    conn.execute(
        "CREATE INDEX IF NOT EXISTS idx_agent_conversations_agent_id ON agent_conversations(agent_id)"
    )
